File: Examples/VisualBasic/WorkingWithResourceAssignments/GetResourceAssignmentCosts.vb

Namespace WorkingWithResourceAssignments
    Public Class GetResourceAssignmentCosts
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:GetResourceAssignmentCosts
            ' Create project instance
            Dim project1 As New Project(dataDir + "ResourceAssignmentCosts.mpp")

            ' Print resource assignment costs
            For Each ra As ResourceAssignment In project1.ResourceAssignments
                Console.WriteLine(ra.[Get](Asn.Cost))
                Console.WriteLine(ra.[Get](Asn.ACWP))
                Console.WriteLine(ra.[Get](Asn.BCWP))
                Console.WriteLine(ra.[Get](Asn.BCWS))
            Next
            ' ExEnd:GetResourceAssignmentCosts
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithResourceAssignments/GetResourceAssignmentCosts.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithResourceAssignments
{
    using System;

    internal class GetResourceAssignmentCosts
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:GetResourceAssignmentCosts
            //ExFor: Asn.Cost
            //ExFor: Asn.ACWP
            //ExFor: Asn.BCWP
            //ExFor: Asn.BCWS
            //ExSummary: Shows how to read assignment's costs.
            var project = new Project(dataDir + "ResourceAssignmentCosts.mpp");

            // Print resource assignment costs
            foreach (var ra in project.ResourceAssignments)
            {
                Console.WriteLine(ra.Get(Asn.Cost));
                Console.WriteLine(ra.Get(Asn.ACWP));
                Console.WriteLine(ra.Get(Asn.BCWP));
                Console.WriteLine(ra.Get(Asn.BCWS));
            }
            //ExEnd:GetResourceAssignmentCosts
        }
    }
}
